### ISSUE TITLE
bug(bigquery): segmentation fault when downloading data on macOS

### ISSUE THREAD

**Issue by openerror** [ibis-project/ibis#6951]
### What happened?

The same code and library versions worked a few weeks prior. But now whenever I call the `.execute()` or `to_*()` methods of a table relations object, Python crashes and Terminal reports a segmentation fault. The relations object would actually compile and give me the SQL sent to BigQuery---but any attempt at downloading data, as mentioned, crashes Python itself.

Tried ibis 5.1.0/6.1.0 + macOS Ventura 13.5.1 and ibis 6.1.0 + macOS Monterey 12.6.1; encountered the same problem. Wondering if anyone is seeing the same?

### What version of ibis are you using?

5.1.0 and 6.1.0. I have made sure to test things in freshly created conda environments; the only set up I did was `mamba install -c conda-forge -y ibis-bigquery google-cloud-storage`.

### What backend(s) are you using, if any?

Google BigQuery

### Relevant log output

```sh
No error messages are printed except the seg fault itself.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by openerror** [comment]
I have isolated and solved the problem, which has two components: pip versus conda, and bigquery permissions.

The seg fault disappeared after I had switched to the `pip` version of ibis. It was only then the proper error message was printed (permissions related). Once the proper permissions were acquired, the same code began to work for both `conda` and `pip` versions of ibis.

It is unfortunate that conda version is somehow less...robust? On that note, earlier when my code and permissions were working properly, conda ibis would print the following warning message whenever I try to download any BigQuery data. Seems like there weird memory/IO/networking issues to begin with.

`E0829 11:38:59.678314000 140704493758208 client_channel.cc:667]        chand=0x7f79a5a93460: Illegal keepalive throttling value `
